nexus/tests/integration_tests/external_ips.rs

```diff
@@ -249,6 +252,128 @@ async fn test_floating_ip_create(cptestctx: &ControlPlaneTestContext) {
+    // silo, which is what the helper would do
+    let _ = create_ip_pool(&client, "default", None).await;
+    link_ip_pool(&client, "default", &silo.identity.id, true).await;
+
+    // create other pool and link to silo
+    let other_pool_range = IpRange::V4(
+        Ipv4Range::new(Ipv4Addr::new(10, 1, 0, 1), Ipv4Addr::new(10, 1, 0, 5))
+            .unwrap(),
+    );
+    create_ip_pool(&client, "other-pool", Some(other_pool_range)).await;
+    link_ip_pool(&client, "other-pool", &silo.identity.id, false).await;
+
+    // create third pool and don't link to silo
+    let unlinked_pool_range = IpRange::V4(
+        Ipv4Range::new(Ipv4Addr::new(10, 2, 0, 1), Ipv4Addr::new(10, 2, 0, 5))
+            .unwrap(),
+    );
+    create_ip_pool(&client, "unlinked-pool", Some(unlinked_pool_range)).await;
+
+    // Create a silo user
+    let user = create_local_user(
+        client,
+        &silo,
+        &"user".parse().unwrap(),
+        params::UserPassword::LoginDisallowed,
+    )
+    .await;
+
+    // Make silo collaborator
+    grant_iam(
+        client,
+        &silo_url,
+        SiloRole::Collaborator,
+        user.id,
+        AuthnMode::PrivilegedUser,
+    )
+    .await;
+
+    // create project as user (i.e., in their silo)
+    NexusRequest::objects_post(
+        client,
+        "/v1/projects",
+        &params::ProjectCreate {
+            identity: IdentityMetadataCreateParams {
+                name: PROJECT_NAME.parse().unwrap(),
+                description: "floating ip project".to_string(),
+            },
+        },
+    )
+    .authn_as(AuthnMode::SiloUser(user.id))
+    .execute()
+    .await
+    .expect("Failed to create project");
+
+    let create_url = get_floating_ips_url(PROJECT_NAME);
+
+    // create a floating IP as this user, first with default pool
+    let body = params::FloatingIpCreate {
+        identity: IdentityMetadataCreateParams {
+            name: "root-beer".parse().unwrap(),
+            description: String::from("a floating ip"),
+        },
+        pool: None,
+        ip: None,
+    };
+    let fip: views::FloatingIp =
+        NexusRequest::objects_post(client, &create_url, &body)
+            .authn_as(AuthnMode::SiloUser(user.id))
+            .execute_and_parse_unwrap()
+            .await;
+    assert_eq!(fip.identity.name.to_string(), "root-beer");
+
+    // now with other pool linked to my silo
+    let body = params::FloatingIpCreate {
+        identity: IdentityMetadataCreateParams {
+            name: "another-soda".parse().unwrap(),
+            description: String::from("a floating ip"),
+        },
+        pool: Some(NameOrId::Name("other-pool".parse().unwrap())),
+        ip: None,
+    };
+    let fip: views::FloatingIp =
+        NexusRequest::objects_post(client, &create_url, &body)
+            .authn_as(AuthnMode::SiloUser(user.id))
+            .execute_and_parse_unwrap()
+            .await;
+    assert_eq!(fip.identity.name.to_string(), "another-soda");
```

Comment:
This request fails as expected without the fix.

Reply:
I do really like this test for running through (default, linked, unlinked) pools on a more typical user. Thanks for putting it together.